### ISSUE TITLE
Better Yapping Functionality

### ISSUE THREAD

**Issue by Justiniscoding** [stratosvomvos/YapsLock#1]
I'm assuming this was written in CPP but not sure. If this was, then what I'm describing is possible :)

- Have the app run in the background, not in a terminal window
- Have it detect when the ~~caps~~ yaps lock key is pressed
- Let the user type something, which the application will record
- When the user presses enter, the app gets yapping :)

The advantage of this is that it can yap directly into a word document or some other app for an amazing essay that will never end 😉

If you put the code on here (which I couldn't see, but I might be stupid) I could help add these features if you want

**Comment by chez-main** [comment]
Yeah, this is just batch and the compiled version of Lammacpp. Im currently working on a python rewrite (using the lammacpp-python library), thanks for the recommendation!

**Comment by Justiniscoding** [comment]
> Yeah, this is just batch and the compiled version of Lammacpp. Im currently working on a python rewrite (using the lammacpp-python library), thanks for the recommendation!

I'll try to implement this in JavaScript with node.js, I know how to do everything except the AI part (shouldn't be too hard… I hope). Is there any alternative to forcing the user to install the model locally? Is there some sort of free API?

**Comment by chez-main** [comment]
> > Yeah, this is just batch and the compiled version of Lammacpp. Im currently working on a python rewrite (using the lammacpp-python library), thanks for the recommendation!
> 
> I'll try to implement this in JavaScript with node.js, I know how to do everything except the AI part (shouldn't be too hard… I hope). Is there any alternative to forcing the user to install the model locally? Is there some sort of free API?

I am going to use langchain cus it runs locally, doesn't take too much storage and it has a python api (kind of), i thought of using the huggingface api to make this in javascript and html5 but i dont like sending stuff in a server in china or smth and its just as fast.
